server/src/main/java/org/elasticsearch/index/translog/Checkpoint.java

```diff
@@ -101,26 +120,49 @@ private void write(DataOutput out) throws IOException {
         out.writeLong(maxSeqNo);
         out.writeLong(globalCheckpoint);
         out.writeLong(minTranslogGeneration);
+        out.writeLong(trimmedAboveSeqNo);
     }
 
     static Checkpoint emptyTranslogCheckpoint(final long offset, final long generation, final long globalCheckpoint,
                                               long minTranslogGeneration) {
         final long minSeqNo = SequenceNumbers.NO_OPS_PERFORMED;
         final long maxSeqNo = SequenceNumbers.NO_OPS_PERFORMED;
-        return new Checkpoint(offset, 0, generation, minSeqNo, maxSeqNo, globalCheckpoint, minTranslogGeneration);
+        return new Checkpoint(offset, 0, generation, minSeqNo, maxSeqNo, globalCheckpoint, minTranslogGeneration, maxSeqNo);
```

Comment:
can we use SequenceNumbers.UNASSIGNED_SEQ_NO? then we can see if things were trimmed or not.